tests/roundrobin_poll_test.cpp

```diff
@@ -0,0 +1,162 @@
+}
+
+static Configuration make_consumer_config(const string& group_id = "rr_consumer_test") {
+    Configuration config;
+    config.set("metadata.broker.list", KAFKA_TEST_INSTANCE);
+    config.set("enable.auto.commit", true);
+    config.set("enable.auto.offset.store", true );
+    config.set("auto.commit.interval.ms", 100);
+    config.set("group.id", group_id);
+    return config;
+}
+
+static vector<int> make_roundrobin_partition_vector(int total_messages) {
+    vector<int> partition_order;
+    for (int i = 0, partition = 0; i < total_messages+1; ++i) {
+        if ((i % KAFKA_NUM_PARTITIONS) == 0) {
+            partition = 0;
+        }
+        partition_order.push_back(partition++);
+    }
+    return partition_order;
+}
+
+//========================================================================
+//                              TESTS
+//========================================================================
+
+TEST_CASE("serial consumer test", "[roundrobin consumer]") {
+    int messages_per_partition = 3;
+    int total_messages = KAFKA_NUM_PARTITIONS * messages_per_partition;
+
+    // Create a consumer and subscribe to the topic
+    Consumer consumer(make_consumer_config());
+    TopicPartitionList partitions;
+    for (int i = 0; i < KAFKA_NUM_PARTITIONS; partitions.emplace_back(KAFKA_TOPICS[0], i++));
+    consumer.assign(partitions);
+    
+    // Start the runner with the original consumer
+    ConsumerRunner runner(consumer, total_messages, KAFKA_NUM_PARTITIONS);
+
+    // Produce messages so we stop the consumer
+    Producer producer(make_producer_config());
+    string payload = "Serial";
+    
+    // push 3 messages in each partition
+    for (int i = 0; i < total_messages; ++i) {
+        producer.produce(MessageBuilder(KAFKA_TOPICS[0]).partition(i%KAFKA_NUM_PARTITIONS).payload(payload));
+    }
+    producer.flush();
+    runner.try_join();
+    
+    // Check that we have all messages
+    REQUIRE(runner.get_messages().size() == total_messages);
+    
+    // messages should have sequential identical partition ids in groups of <messages_per_partition>
+    int expected_partition;
+    for (int i = 0; i < total_messages; ++i) {
+        if ((i % messages_per_partition) == 0) {
+            expected_partition = runner.get_messages()[i].get_partition();
+        }
+        REQUIRE(runner.get_messages()[i].get_partition() == expected_partition);
+        REQUIRE((string)runner.get_messages()[i].get_payload() == payload);
+    }
+}
+
+TEST_CASE("roundrobin consumer test", "[roundrobin consumer]") {
+    TopicPartitionList assignment;
+    int messages_per_partition = 3;
+    int total_messages = KAFKA_NUM_PARTITIONS * messages_per_partition;
+
+    // Create a consumer and subscribe to the topic
+    PollStrategyAdapter consumer(make_consumer_config());
+    TopicPartitionList partitions;
+    for (int i = 0; i < KAFKA_NUM_PARTITIONS; partitions.emplace_back(KAFKA_TOPICS[0], i++));
+    consumer.assign(partitions);
```

Comment:
Any reason why you're not using `subscribe` here? That involves a slightly trickier code flow so it may be worth using it instead.

Reply:
Ok will add subscribe.